app/controllers/admin/lead_providers_controller.rb

```diff
@@ -1,42 +1,118 @@
 # frozen_string_literal: true
 
 class Admin::LeadProvidersController < Admin::BaseController
-  skip_after_action :verify_authorized, only: :index
-  skip_after_action :verify_policy_scoped, except: :index
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
 
-  def index
-    @lead_providers = policy_scope(LeadProvider)
+  def choose_cip
+    authorize LeadProvider, :create?
+
+    new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    if session[:lead_provider_form].nil?
+      session[:lead_provider_form] = { name: new_supplier_form.name }
+    else
+      session[:lead_provider_form].merge!({ name: new_supplier_form.name })
+    end
+
+    @lead_provider_form = LeadProviderForm.new(session[:lead_provider_form])
+  end
+
+  def receive_cip
+    authorize LeadProvider, :create?
+
+    if params.dig(:lead_provider_form, :cip).blank?
+      @lead_provider_form = LeadProviderForm.new
+      @lead_provider_form.errors.add(:cip, :blank, message: "Choose one")
+      render :choose_cip and return
+    end
+
+    session[:lead_provider_form].merge!(params.require(:lead_provider_form).permit(:cip))
+    redirect_to admin_new_lead_provider_cohorts_path
   end
 
-  def new
-    @lead_provider = LeadProvider.new
-    authorize @lead_provider
+  def choose_cohorts
+    authorize LeadProvider, :create?
+    @lead_provider_form = LeadProviderForm.new(session[:lead_provider_form])
+  end
+
+  def receive_cohorts
+    authorize LeadProvider, :create?
+    cohorts = params.dig(:lead_provider_form, :cohorts)
+                &.keep_if(&:present?)
+
+    if cohorts.blank?
+      @lead_provider_form = LeadProviderForm.new
+      @lead_provider_form.errors.add(:cohorts, :blank, message: "Choose one or more")
+      render :choose_cohorts and return
+    end
+
+    session[:lead_provider_form].merge!({ cohorts: cohorts })
+    redirect_to admin_new_lead_provider_review_path
+  end
+
+  def review
+    authorize LeadProvider, :create?
+
+    @lead_provider_form = LeadProviderForm.new(session[:lead_provider_form])
   end
 
   def create
-    @lead_provider = LeadProvider.new(permitted_attributes(LeadProvider))
-    authorize @lead_provider
+    @lead_provider_form = LeadProviderForm.new(session[:lead_provider_form])
 
-    if @lead_provider.save
-      redirect_to admin_lead_providers_path
-    else
-      render :new
+    @lead_provider = LeadProvider.new(name: @lead_provider_form.name, cohorts: @lead_provider_form.chosen_cohorts)
+    authorize @lead_provider, :create?
+
+    ActiveRecord::Base.transaction do
+      @lead_provider.save!
+      @lead_provider_form.chosen_cohorts.each do |cohort|
+        LeadProviderCip.create!(cohort: cohort, cip: @lead_provider_form.chosen_cip, lead_provider: @lead_provider)
+      end
     end
```

Comment:
Like with the DP form, I would be tempted to have a save method there that deals with creating a lead provider, and its needed relations.